### ISSUE TITLE
Numeric object property names fail in jsdoc@3.3.0-beta2

### ISSUE THREAD

**Issue by lazd** [jsdoc3/jsdoc#945]
Something like this:

```js
  /**
    Used to check if a key is a modifier.
    @ignore
  */
  var modifierCodeMap = {
    16: true,
    17: true,
    18: true,
    91: true
  };
```

Causes this:

```
/node_modules/jsdoc/lib/jsdoc/tag.js:33
    return meta && meta.code && meta.code.name === text && text.match(/(?:^\s+
                                                                ^
TypeError: Object 16 has no method 'match'
    at mustPreserveWhitespace (/node_modules/jsdoc/lib/jsdoc/tag.js:33:65)
    at trim (/node_modules/jsdoc/lib/jsdoc/tag.js:43:10)
    at new exports.Tag (/node_modules/jsdoc/lib/jsdoc/tag.js:189:17)
    at Doclet.addTag (/node_modules/jsdoc/lib/jsdoc/doclet.js:201:18)
    at newSymbolDoclet (/node_modules/jsdoc/lib/jsdoc/src/handlers.js:256:19)
    at null.<anonymous> (/node_modules/jsdoc/lib/jsdoc/src/handlers.js:318:13)
    at EventEmitter.emit (events.js:98:17)
    at Visitor.visitNode (/node_modules/jsdoc/lib/jsdoc/src/visitor.js:281:16)
    at Visitor.visit (/node_modules/jsdoc/lib/jsdoc/src/visitor.js:191:27)
    at Walker.recurse (/node_modules/jsdoc/lib/jsdoc/src/walker.js:531:44)
```

Because [this line](https://github.com/jsdoc3/jsdoc/blob/65f307322ac64d965694dca8f82bf98f08a1e2e4/lib/jsdoc/tag.js#L33) assumes that `text` will always be a string.

A simple `toString()` [right about here](https://github.com/jsdoc3/jsdoc/blob/65f307322ac64d965694dca8f82bf98f08a1e2e4/lib/jsdoc/tag.js#L189) would do the trick.

This problem was not present in jsdoc@3.3.0-beta1.

**Comment by ergose** [comment]
I think I have the same problem. I just installed it globally on my linux box with ` sudo npm install jsdoc@"<=3.3.0" `, then I created a directory named "codedoc" and ran ` jsdoc ./jade.js -d ./codedoc/ `. 

This is what I'm getting:
```
/usr/local/lib/node_modules/jsdoc/lib/jsdoc/tag.js:33
    return meta && meta.code && meta.code.name === text && text.match(/(?:^\s+
                                                                ^
TypeError: Object 1 has no method 'match'
    at mustPreserveWhitespace (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/tag.js:33:65)
    at trim (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/tag.js:43:10)
    at new exports.Tag (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/tag.js:189:17)
    at Doclet.addTag (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/doclet.js:199:18)
    at newSymbolDoclet (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/handlers.js:256:19)
    at null.<anonymous> (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/handlers.js:318:13)
    at EventEmitter.emit (events.js:98:17)
    at Visitor.visitNode (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/visitor.js:281:16)
    at Visitor.visit (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/visitor.js:191:27)
    at Walker.recurse (/usr/local/lib/node_modules/jsdoc/lib/jsdoc/src/walker.js:531:44)
```

**Comment by lazd** [comment]
@ergose that is definitely the same issue. I'm putting together a test and PR now!

**Comment by ergose** [comment]
@lazd Thanks for mentioning that it wasn't in beta1. I rolled back to it and was able to get the documentation I was needing. :) Once the PR goes through then I'll update back to the beta2.

**Comment by lazd** [comment]
@ergose I submitted a PR that includes a failing test, but my simple fix wasn't enough to get JSDoc to work with this (see PR for details). Hopefully @hegemonic knows exactly what to do given the test and insights in the PR :)

**Comment by hegemonic** [comment]
Fixed on master and releases/3.3. Thanks for the test case!

**Comment by lazd** [comment]
You're welcome!
